src/query/service/src/interpreters/util.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub fn format_name(name: &str, quoted_ident_case_sensitive: bool) -> String{
+    // Db-s -> "Db-s" ; dbs -> dbs
+    if name.chars().find(|c| !(c.is_lowercase() || c.is_numeric() || *c == '_') ).is_some() && quoted_ident_case_sensitive {
+        format!("\"{name}\"")
```

Comment:
Not every dialect uses `"` as ident qoute

Reply:
MySQL use `, other use "?

Reply:
most dialect support `"` , but only mysql supports   `

Reply:
PostgreSQL uses \` while mysql uses `"`

Reply:
I suggest reading the quote from dialect and carefully escape the name by the selected quote.

Reply:
> I suggest reading the quote from dialect and carefully escape the name by the selected quote.
> 
> FYI, there is a `quote_ident()` function.

in interpreter show create table , we can not get the string quote. Only the name.

Reply:
> we can not get the string quote

We can select the default quote used by the dialect.